### DISCUSSION TITLE
Custom fullscreen in ScrollView

### DISCUSSION THREAD

**Discussion by bartoszchlad** [TheWidlarzGroup/react-native-video#4951] [Ideas]
### What problem are you trying to solve?

I created a player with custom controls and I have a miniplayer with the ability to go fullscreen. However, going fullscreen only works if the player is not inside a `ScrollView`, or if I open fullscreen in a modal — but opening in a modal causes the video buffer to be lost.

### Proposed solution

A way to have a custom miniplayer with the ability to go fullscreen while the player is inside a `ScrollView`, without losing the video buffer.

### Alternatives considered

- Placing the player outside a `ScrollView` (custom fullscreen works, but doesn't fit the desired layout).
- Opening fullscreen in a modal (works, but causes losing the buffer of the video).

### Additional context

Is this even possible to have a custom miniplayer with the ability to go fullscreen in a `ScrollView` with react-native-video?

**Comment by freeboub** [comment]
Did you check this project ? https://github.com/Lg0gs/react-native-corner-video

It looks like similar use case.

Let me know if it helps!

**Comment by bartoszchlad** [comment]
Unfortunately, neither the library nor the solution used in it solves the problem, because we still lose the video buffer as in the case of the solution using a modal :(

**Comment by freeboub** [comment]
Ok, in my app, I declare video player outside of the react-native-navigation stack. It allows us to keep video player available during app browsing. I think you need to implement such navigation trick

[a comment by an automated account is not shown]

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
